R/check_dates.R

```diff
@@ -14,6 +14,10 @@ check_dates <- function(x, error_on_NA = FALSE, ...) {
     stop("dates is NULL", call. = FALSE)
   }
 
+  if (is.character(x)) {
+    x <- as.Date(x, ...)
```

Comment:
So the reason why we removed that was some nasty behaviour by `as.Date`, e.g. 
```
> as.Date("01-01-2001")
[1] "1-01-20"
```
It looks like this issue will remain here.

Reply:
``` r
incidence::incidence("02-01-2001")
#> Error in incidence.character("02-01-2001"): Not all dates are in ISO 8601 standard format (yyyy-mm-dd). The first incorrect date is 02-01-2001
```

<sup>Created on 2019-01-12 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>